site/src/helpers/courseRequirements.ts

```diff
@@ -129,12 +129,38 @@ export function collapseSingletonRequirements(requirements: ProgramRequirement[]
   return computedRequirements;
 }
 
-export function flattenSingletonGroups(requirements: ProgramRequirement[]): ProgramRequirement[] {
+/**
+ * Flattens requirements to prevent deep nesting.
+ * Preserves the titles of the top-level categories by default.
+ *
+ * ```typescript
+ *
+ *   requirements = flattenSingletonGroups(requirements);
+ *
+ * ```
```

Comment:
I don't see a reason to include this comment; usage from the function signature should be self-explanatory
```suggestion
```